lib/shoryuken/fetcher.rb

```diff
@@ -9,42 +9,30 @@ def initialize(manager)
       @manager = manager
     end
 
-    def receive_messages(queue, limit)
-      # AWS limits the batch size by 10
-      limit = limit > FETCH_LIMIT ? FETCH_LIMIT : limit
-
-      options = (Shoryuken.options[:aws][:receive_message] || {}).dup
-      options[:max_number_of_messages] = limit
-      options[:message_attribute_names] = %w(All)
-      options[:attribute_names] = %w(All)
-
-      Shoryuken::Client.queues(queue).receive_messages options
-    end
-
     def fetch(queue, available_processors)
       watchdog('Fetcher#fetch died') do
         started_at = Time.now
 
         logger.debug { "Looking for new messages in '#{queue}'" }
 
         begin
-          batch = Shoryuken.worker_registry.batch_receive_messages?(queue)
+          batch = Shoryuken.worker_registry.batch_receive_messages?(queue.name)
```

Comment:
If I remember correctly, it's a long time ago now, this line was maintaining the interface. The refactor mean that `fetch` received a queue object (`QueueConfiguration` by the looks of it) rather than just a `String` and so for downstream calls I had to pass `queue.name` so that they received a `String` as before and therefore didn't need to change.